### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by pinkmoon25**
### Blog app - Add forms

- Create current_user method in Application controller
- Create form to create posts on behalf of a specific user
- Create form to create comments on behalf of a specific user
- Allow Users to add likes to Posts.
- Configure Routes for new and create actions
- Implement new and create method for posts controller
- Implement create methods for comment and likes controller

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,5 +6,25 @@ def index
...
     @author = User.find(params[:user_id])
     @post = @author.posts.find((params[:id]))
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:15]: - [ ] OPTIONAL
It's good rails practice to use _strong parameters_ for query parameters, which should be similar to:

```ruby
# controllers/posts_controller.rb

  class PostsController < ApplicationController

       ...
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end

      ...

  end
```

Kindly read further lesson on  strong parameters via [this link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CommentsController < ApplicationController
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:1]: - [ ] `new` action is missing in your comments controller. Kindly note that this module requires that you explicitly declare both `new` and `create` actions for comments controller. Kindly ensure to improve upon the same, which should be similar to:
```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

  def create
  
    ...
  end


  private 
  def comment_params
    params.require(:comment).permit(:text)
  end

	...
end

File: config/routes.rb [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-  resources :users, only: [:index, :show] do 
-    resources :posts, only: [:index, :show]
-  end
+  get "/users", to: 'users#index', as: 'users'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_recent_posts'
+  post '/users/:user_id/posts', to: 'posts#create'
+  get '/users/:user_id/posts/new', to: 'posts#new', as: 'new_post'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_all_posts'
+  post '/users/:user_id/posts/:id', to: 'comments#create'
```
Comment (rloterh) [inline config/routes.rb:9]: - [ ] As per the comments controller actions `new` and `create`, kindly ensure that there is a pair of corresponding routes in the `config/routes.rb` file for creating comments, which should yield:
```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Tushar, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good progress with form implementation

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by sole contributor. 
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by pinkmoon25** [comment]
I have implemented the required changes

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @PinkMoon25 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**